pkg/installmanager/dnscleanup.go

```
@@ -1,56 +1,72 @@
 // types may be added in the future, but right now this is the only one we're seeing
 // leak and conflict.
 // May no longer be necessary once https://jira.coreos.com/browse/CORS-1195 is fixed.
-func cleanupDNSZone(dnsZoneID, region string, logger log.FieldLogger) error {
-	zoneLogger := logger.WithField("dnsZoneID", dnsZoneID)
+func cleanupAWSDNSZone(dnsZone *hivev1.DNSZone, region string, logger log.FieldLogger) error {
+	zoneLogger := logger.WithField("dnsZoneID", *dnsZone.Status.AWS.ZoneID)
 	zoneLogger.Info("cleaning up DNSZone")
 
 	awsClient, err := awsclient.NewClient(nil, "", "", region)
 	if err != nil {
 		return err
 	}
-	recordsOutput, err := awsClient.ListResourceRecordSets(
-		&route53.ListResourceRecordSetsInput{
-			HostedZoneId: awssdk.String(dnsZoneID),
-		},
-	)
-	if err != nil {
-		return err
+
+	recordTypes := []string{
+		route53.RRTypeA,
 	}
-	for _, r := range recordsOutput.ResourceRecordSets {
-		// We're only experiencing problems with A records, so these are all we cleanup for now:
-		if *r.Type == "A" {
-			zoneLogger.WithFields(log.Fields{"name": *r.Name, "type": *r.Type}).Info("deleting A record")
-			request := &route53.ChangeResourceRecordSetsInput{
-				ChangeBatch: &route53.ChangeBatch{
-					Changes: []*route53.Change{
-						{
-							Action:            awssdk.String("DELETE"),
-							ResourceRecordSet: r,
-						},
-					},
-				},
-				HostedZoneId: awssdk.String(dnsZoneID),
-			}
-			_, err := awsClient.ChangeResourceRecordSets(request)
-			if err != nil {
-				logger.WithError(err).WithField("recordset", r.Name).Warn("error deleting recordset")
-				return err
-			}
-		}
+	if err := dns.DeleteAWSRecordSets(awsClient, dnsZone, recordTypes, zoneLogger); err != nil {
+		logger.WithError(err).Error("failed to clean up DNS Zone")
+	} else {
+		zoneLogger.Info("DNSZone A records deleted")
 	}
-	zoneLogger.Info("DNSZone A records deleted")
-	return nil
+	return err
```

I think that you want to return the error returned from `dns.DeleteAWSRecordSets`. This is returning the error returned from `awsclient.NewClient`, which at this point will always be nil.